062-support-tiered-storage.md

```diff
@@ -0,0 +1,92 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for `RemoteStorageManager`, users are able to adopt tiered storage feature on Strimzi easily.
+
+For simplicity, the proposal doesn't add support to customize `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation class `TopicBasedRemoteLogMetadataManager` in Kafka will be used.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
```

Comment:
Does it mean that users are required to build their custom images based on Strimzi Kafka images? Can you provide an example? We should also consider having an automatism like we have for Kafka Connect, where you can specify plugins URLs in the configuration and the image is built for you.

Reply:
Yes in our case we customize the docker image based on Strimzi built image. The only customization we add is to copy the plugin library jar file into a plugin path. I include a section with an example dockerfile to build such image.

Reply:
I'm fine with that. I don't know if we want to mention the automated build as rejected alternative with the motivation expressed by Jakub. It's up to you.